From e7e2d1929a03dfdb0751343708d3311db0a9c400 Mon Sep 17 00:00:00 2001
From: ole <gitlab+account@ole.mn>
Date: Sat, 10 Dec 2022 07:04:58 +0100
Subject: [PATCH] theme template

---
 src/kgx-terminal.c | 64 ++++++++++++++++++++++++++++++----------------
 1 file changed, 42 insertions(+), 22 deletions(-)

diff --git a/src/kgx-terminal.c b/src/kgx-terminal.c
index 26ca81d..85339f0 100644
--- a/src/kgx-terminal.c
+++ b/src/kgx-terminal.c
@@ -113,24 +113,42 @@ update_terminal_colors (KgxTerminal *self)
   GdkRGBA fg;
   GdkRGBA bg;
 
-  // Workings of GDK_RGBA prevent this being static
-  GdkRGBA palette[16] = {
-    GDK_RGBA ("241f31"), // Black
-    GDK_RGBA ("c01c28"), // Red
-    GDK_RGBA ("2ec27e"), // Green
-    GDK_RGBA ("f5c211"), // Yellow
-    GDK_RGBA ("1e78e4"), // Blue
-    GDK_RGBA ("9841bb"), // Magenta
-    GDK_RGBA ("0ab9dc"), // Cyan
-    GDK_RGBA ("c0bfbc"), // White
-    GDK_RGBA ("5e5c64"), // Bright Black
-    GDK_RGBA ("ed333b"), // Bright Red
-    GDK_RGBA ("57e389"), // Bright Green
-    GDK_RGBA ("f8e45c"), // Bright Yellow
-    GDK_RGBA ("51a1ff"), // Bright Blue
-    GDK_RGBA ("c061cb"), // Bright Magenta
-    GDK_RGBA ("4fd2fd"), // Bright Cyan
-    GDK_RGBA ("f6f5f4"), // Bright White
+  GdkRGBA palette[16];
+  GdkRGBA light_palette[16] = {
+    GDK_RGBA ("3b4252"), // Black
+    GDK_RGBA ("bf616a"), // Red
+    GDK_RGBA ("a3be8c"), // Green
+    GDK_RGBA ("ebcb8b"), // Yellow
+    GDK_RGBA ("5e81ac"), // Blue
+    GDK_RGBA ("b48ead"), // Magenta
+    GDK_RGBA ("88c0d0"), // Cyan
+    GDK_RGBA ("d8dee9"), // White
+    GDK_RGBA ("4c566a"), // Bright Black
+    GDK_RGBA ("bf616a"), // Bright Red
+    GDK_RGBA ("a3be8c"), // Bright Green
+    GDK_RGBA ("ebcb8b"), // Bright Yellow
+    GDK_RGBA ("81a1c1"), // Bright Blue
+    GDK_RGBA ("b48ead"), // Bright Magenta
+    GDK_RGBA ("8fbcbb"), // Bright Cyan
+    GDK_RGBA ("e5e9f0"), // Bright White
+  };
+  GdkRGBA dark_palette[16] = {
+    GDK_RGBA ("3B4252"), // Black
+    GDK_RGBA ("BF616A"), // Red
+    GDK_RGBA ("A3BE8C"), // Green
+    GDK_RGBA ("EBCB8B"), // Yellow
+    GDK_RGBA ("81A1C1"), // Blue
+    GDK_RGBA ("B48EAD"), // Magenta
+    GDK_RGBA ("88C0D0"), // Cyan
+    GDK_RGBA ("E5E9F0"), // White
+    GDK_RGBA ("4C566A"), // Bright Black
+    GDK_RGBA ("BF616A"), // Bright Red
+    GDK_RGBA ("A3BE8C"), // Bright Green
+    GDK_RGBA ("EBCB8B"), // Bright Yellow
+    GDK_RGBA ("81A1C1"), // Bright Blue
+    GDK_RGBA ("B48EAD"), // Bright Magenta
+    GDK_RGBA ("8FBCBB"), // Bright Cyan
+    GDK_RGBA ("ECEFF4"), // Bright White
   };
 
   if (self->theme == KGX_THEME_AUTO) {
@@ -151,14 +169,16 @@ update_terminal_colors (KgxTerminal *self)
       bg = (GdkRGBA) { 0.05, 0.05, 0.05, 0.96 };
       break;
     case KGX_THEME_DAY:
-      fg = (GdkRGBA) { 0.0, 0.0, 0.0, 0.0 };
-      bg = (GdkRGBA) { 1.0, 1.0, 1.0, 1.0 };
+      fg = (GdkRGBA) { 0.180, 0.204, 0.251, 1.00 };
+      bg = (GdkRGBA) { 0.976, 0.980, 0.984, 0.8 };
+      memcpy(palette, light_palette, sizeof(light_palette));
       break;
     case KGX_THEME_NIGHT:
     case KGX_THEME_AUTO:
     default:
-      fg = (GdkRGBA) { 1.0, 1.0, 1.0, 1.0};
-      bg = (GdkRGBA) { 0.05, 0.05, 0.05, 0.96 };
+      fg = (GdkRGBA) { 0.847, 0.871, 0.914, 1.00 };
+      bg = (GdkRGBA) { 0.180, 0.204, 0.251, 1.0 };
+      memcpy(palette, dark_palette, sizeof(dark_palette));
       break;
   }
 
-- 
2.38.1
